.github/CONTRIBUTING.md

```diff
@@ -1,42 +1,251 @@
 # Contributing to Rocket.Chat
 
-:+1::tada: First off, thanks for taking the time to contribute! :tada::+1:
+**First off, thanks for taking the time to contribute! :tada::+1:**
 
-The following is a set of guidelines for contributing to Rocket.Chat and its packages, which are hosted in the [Rocket.Chat Organization](https://github.com/RocketChat) on GitHub.
+> There are many ways to contribute to Rocket.Chat even if you're not technical or a developer:
+>
+> * Email us at marketing@rocket.chat to tell us how much you love the project
+> * Write about us in your blogs
+> * Fix some small typos in our [documentation](https://github.com/RocketChat/docs)
+> * Become our [GitHub sponsor](https://github.com/sponsors/RocketChat)
+> * Tell others about us and help us spread the word
+>
+> Every bit of contribution is appreciated 🙂 thank you!
+
+The following is a set of guidelines for contributing to Rocket.Chat, which are hosted in the [Rocket.Chat Organization](https://github.com/RocketChat) on GitHub.
 
 __Note:__ If there's a feature you'd like, there's a bug you'd like to fix, or you'd just like to get involved please raise an issue and start a conversation. We'll help as much as we can so you can get contributing - although we may not always be able to respond right away :)
 
-## ECMAScript 2015 vs CoffeeScript
+## Setup
+
+Rocket.Chat runs on top of [Meteor](https://www.meteor.com/). To run it on development mode you need to [install Meteor](https://www.meteor.com/install) and clone/download the Rocket.Chat's code, then just open the code folder and run:
+```shell
+meteor npm install && meteor
+```
+It should build and run the application and database for you, now you can access the UI on (http://localhost:3000)
+
+It's not necessary to install Nodejs or NPM, every time you need to use them you can run `meteor node` or `meteor npm`.
+
+It's important to always run the NPM commands using `meteor npm` to ensure that you are installing the modules using the right Nodejs version.
+
+## Coding
+
+We provide a [.editorconfig](../.editorconfig) file that will help you to keep some standards in place.
+
+### ECMAScript vs TypeScript
+
+We are currently adopting TypeScript as the default language on our projects, the current codebase will be migrated, along the time, from JavaScript to TypeScript.
+
+While we still have a lot of JavaScript files you should not create new ones. As much as possible new code contributions should be in **TypeScript**.
+
+### Blaze vs React
 
-While we still have a lot of CoffeeScript files you should not create new ones. New code contributions should be in **ECMAScript 2015**.
+We are currently adopting React over Blaze as our UI engine, the current codebase has being migrated and will continue for some time, so it's possible to find Blaze templates still and some changes or contributions may need to be made in Blaze while we are still evolving our components library.
```

Comment:
Suggested rephrasing:
We are currently adopting React over Blaze as our UI engine, the current codebase is under migration and will continue.  You will still find Blaze templates in our code. Code changes or contributions may need to be made in Blaze while we continue to evolve our components library.